components/gantry/oneaxis/oneaxis.go

```diff
@@ -187,10 +189,79 @@ func (g *oneAxis) Reconfigure(ctx context.Context, deps resource.Dependencies, c
 			return err
 		}
 	}
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	g.cancelFunc = cancelFunc
+	g.checkHit(ctx)
 
 	return nil
 }
 
+func (g *oneAxis) checkHit(ctx context.Context) {
+	g.activeBackgroundWorkers.Add(1)
+	utils.PanicCapturingGo(func() {
+		defer utils.UncheckedErrorFunc(func() error {
+			return g.motor.Stop(ctx, nil)
+		})
+		defer g.activeBackgroundWorkers.Done()
+		for {
+			if ctx.Err() != nil {
+				return
+			}
+			select {
+			case <-ctx.Done():
+				return
+			default:
+			}
+			func() {
+				for i := 0; i < len(g.limitSwitchPins); i++ {
+					hit, err := g.limitHit(ctx, i)
+					if err != nil {
+						g.logger.Error(err)
+						return
+					}
+					if hit {
+						if err := g.moveAway(ctx, i); err != nil {
+							g.logger.Error(err)
+							return
+						}
+					}
+				}
+			}()
```

Comment:
We've now got an anonymous function that returns nothing, that we call immediately. Just inline the code, and change the returns to continues:
```go
...
case <- ctx.Done():
    return
default:
}

for i := 0; i < len(...) {
    hit, err := g.limitHit(...)
    if err {
        continue
    }
    ...
```

Reply:
why continue? I would think that if there's an error, we would want to stop the function and not continue with the next steps that kind of rely on this one

Reply:
You might be right. I had suggested continue because currently we return from the anonymous function, which continues through the loop (because the return value is ignored). If you wanted to return from the outer function, I'm fine with that, and calling the anonymous function added a bug.

Reply:
changing them to continues instead of returns

Reply:
actually ended up just removing altogether because continue would accomplish the same thing for these if statements

Reply:
The anonymous function is still here, and it does even less now. I'd remove lines 223 and 242, and unindent everything between them.